### ISSUE TITLE
TypeError: Cannot read property 'id' of undefined

### ISSUE THREAD

**Issue by loretoparisi** [Microsoft/BotBuilder#117]
I get this error when there is a `\r\n` in the stdin repeated twice or more:

```shell
botframework/node_modules/botbuilder/lib/Session.js:139
        r.childId = ss.callstack[ss.callstack.length - 1].id;
                                                         ^

TypeError: Cannot read property 'id' of undefined
    at Session.endDialog (/botframework/node_modules/botbuilder/lib/Session.js:139:58)
    at //botframework/node_modules/botbuilder/lib/dialogs/IntentDialog.js:41:25
    at Request._callback (/botframework/node_modules/botbuilder/lib/dialogs/LuisDialog.js:37:17)
    at Request.self.callback (/botframework/node_modules/botbuilder/node_modules/request/request.js:200:22)
    at emitTwo (events.js:87:13)
    at Request.emit (events.js:172:7)
    at Request.<anonymous> (/botframework/node_modules/botbuilder/node_modules/request/request.js:1046:10)
    at emitOne (events.js:82:20)
    at Request.emit (events.js:169:7)
    at IncomingMessage.<anonymous> (/botframework/node_modules/botbuilder/node_modules/request/request.js:973:12)
```

**Comment by Stevenic** [comment]
I'll try to see if I can reproduce the bug. Thanks

**Comment by loretoparisi** [comment]
Ok thanks.
The implementation is very simple, it does nothing more than define a LUIS multi-turn dialog callbacks like

```javascript
var dialog = new builder.LuisDialog(model);
bot.add('/', dialog);

dialog.on('MyIntent',
     [
    function (session, args, next) {
        
        console.log( JSON.stringify( args, undefined, 1) );
   
        session.send('You typed in something...');
    }]);  
dialog.onDefault(builder.DialogAction.send("I didn't not understand."));
```

then listening to stdin

```javascript
bot.listenStdin();
```

and running locally with node.

**Comment by Stevenic** [comment]
I was able to reproduce this and just checked in several fixes as there were a number of issues going on. Fortunately it looks like it would only happen when running from a console window but still good to fix. The fixes will be in the next release I'm planning to push later tonight.

**Comment by loretoparisi** [comment]
@Stevenic 💯

**Comment by Stevenic** [comment]
Should be fixed in v0.9.0
